### PR TITLE
Docs: Veterans use case

### LINKED ISSUE #1587: Update docs site with Veteran use case
Add a new page under the _Use Cases_ section: https://docs.calitp.org/benefits/use-cases/

### PR DESCRIPTION
**Pull request by machikoyasuda**
closes #1587 

- Use case paragraph
- Process diagram. Click https://github.com/cal-itp/benefits/pull/1731/files?short_path=f8c3f59#diff-f8c3f5913b1bb71baaf34ba792e6011ef91b0617c0be5cf8cffdd0a445f528c5 to see diagram illustrated.
- Demonstration: No screenshot GIF yet. Will add it once it's made by Design/Product for the Changelog.

### CODE REVIEW

File: docs/use-cases/veterans.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    Rider->>Benefits: visits site
+    Benefits-->>IdG: identity proofing
+    IdG-->>VA: identity proofing
+    Rider->>VA: enters SSN and ID
+    VA-->>IdG: eligibility verification
+    IdG-->>Benefits: eligibility verification
+    Benefits-->>Littlepay: enrollment start
+    Rider->>Littlepay: enters payment card details
+    Littlepay-->>Benefits: enrollment complete
+```
```
Comment (machikoyasuda) [inline docs/use-cases/veterans.md:26]: @angela-tran @thekaveman Is this characterized correctly? I wasn't sure if or where to add Login.gov to the Process Diagram.

File: docs/use-cases/veterans.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Veterans
+
+Another use case for the Benefits application is for U.S. veterans - people who have served in the active military, naval, or air service, and who were discharged or released therefrom under conditions other than dishonorable. The Benefits application verifies the person's veteran status to confirm eligibility and allows those eligible to enroll their contactless payment card for their transit benefit.
+
+On the back-end, the app uses the Identity Gateway to connect to the [Veteran Confirmation API](https://developer.va.gov/explore/api/veteran-confirmation) to confirm veteran status. On the front-end, users must create or log into their Login.gov account to complete the eligibility verification process.
```
Comment (machikoyasuda) [inline docs/use-cases/veterans.md:5]: I omitted the whole part about how we used to have DS Logon, ID.me, etc. I can add that _when_ we get permission to do so.

**Comment by thekaveman** [comment]
@indexing please coordinate with @machikoyasuda on the diagram you were creating last week, and this documentation for the Veteran use case.

**Comment by machikoyasuda** [comment]
@indexing here's the diagram visually: https://github.com/cal-itp/benefits/pull/1731/files?short_path=f8c3f59#diff-f8c3f5913b1bb71baaf34ba792e6011ef91b0617c0be5cf8cffdd0a445f528c5

and the code that makes it is here: https://github.com/cal-itp/benefits/pull/1731/files#diff-f8c3f5913b1bb71baaf34ba792e6011ef91b0617c0be5cf8cffdd0a445f528c5R9-R26

**Comment by machikoyasuda** [comment]
closing in favor of #1748

**Pull request closed without merging**